core/ipc/rpc/camel-impl/src/main/java/org/opennms/core/rpc/camel/CamelRpcServerRouteManager.java

```diff
@@ -77,7 +76,11 @@ private DynamicRpcRouteBuilder(CamelContext context, MinionIdentity identity, Rp
         }
 
         public String getRouteId() {
-            return "RPC.Server." + module.getId();
+            return "RPC.Server." + module.getId() + "-" + Math.abs(module.hashCode());
```

Comment:
Won't this make it difficult to collect route specific statistics though? Since the MBeans are named using the route id.